File: eden/scm/edenscm/hgext/github/github_repo.py
# Copyright (c) Meta Platforms, Inc. and affiliates.
#
# This software may be used and distributed according to the terms of the
# GNU General Public License version 2.

from edenscm.mercurial import git


def is_github_repo(repo) -> bool:
    """Create or update GitHub pull requests."""
    if not git.isgitpeer(repo):
        return False

    try:
        return repo.ui.paths.get("default", "default-push").url.host == "github.com"
    except AttributeError:  # ex. paths.default is not set
        return False

File: eden/scm/lib/github/src/git_hub_repo.rs
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This software may be used and distributed according to the terms of the
 * GNU General Public License version 2.
 */

use serde::Serialize;

#[derive(Debug, Serialize)]
pub struct GitHubRepo {
    pub owner: String,
    pub name: String,
}
